### PR TITLE
Improve the macOS tray UX

### PR DESCRIPTION
**Pull request by sithlord48**
General:
 - Use QList<QAction*> instead of a std::vector<QAction*>
 - Clean up the `showAndActivate` Logic just show always no matter the previous state
 - Clang-Format the `.m` and `.mm` files ( I had to edit one so i did a commit formatting them all) 
 - Added *.m and *.mm to `scipts/lint-clang.py`

 Update the tray for mac OS
 -  Do not show / hide when the icon is pressed
 -  Show "Bring Window to Front"  in the tray menu (ActionRestore)
 - "Bring Window to Front" works when the app is hidden with ⌘ + H

### CODE REVIEW

File: src/gui/src/MainWindow.cpp [hunk 1]
```diff
@@ -226,7 +226,10 @@ void MainWindow::connectSlots()
 
   connect(&m_WindowSaveTimer, &QTimer::timeout, this, &MainWindow::onWindowSaveTimerTimeout);
 
+// Mac os tray will only show a menu
```
Comment (nbolton) [inline src/gui/src/MainWindow.cpp:229]: Silencio, Bruno!

Comment (sithlord48) [reply src/gui/src/MainWindow.cpp:229]: ... We don't talk about Bruno... :P

File: src/gui/src/MainWindow.cpp [hunk 2]
```diff
@@ -523,9 +526,14 @@ void MainWindow::moveEvent(QMoveEvent *event)
 void MainWindow::open()
 {
 
-  std::vector<QAction *> trayMenu = {ui->m_pActionStartCore, ui->m_pActionStopCore, nullptr, ui->m_pActionQuit};
+  QList<QAction *> trayActions{ui->m_pActionStartCore, ui->m_pActionStopCore, nullptr, ui->m_pActionQuit};
 
-  m_TrayIcon.create(trayMenu);
+#ifdef Q_OS_MAC
+  ui->m_pActionRestore->setText(tr("Bring Window to Front"));
```
Comment (nbolton) [inline src/gui/src/MainWindow.cpp:532]: Is it possible to only enable this menu option if the window is hidden?

Comment (sithlord48) [reply src/gui/src/MainWindow.cpp]: Not in a way that works in all cases mac os does not for example tell the app when the user has used command +H  to hide it

**Review by nbolton — APPROVED** [review]

**Pull request merged**